public/modules/custom/helfi_global_navigation/src/Plugin/rest/resource/MenuLinkCollection.php

```
@@ -0,0 +1,100 @@
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) : self {
+    $instance = parent::create(
+      $container,
+      $configuration,
+      $plugin_id,
+      $plugin_definition
+    );
+    $instance->treeBuilder = $container->get('helfi_navigation.menu_tree_builder');
+
+    return $instance;
+  }
+
+  /**
+   * Callback for GET requests.
+   *
+   * @return \Drupal\rest\ResourceResponse
+   *   The response.
+   */
+  public function get(Request $request): ResourceResponse {
+    if (!$menuName = $request->attributes->get('menu_name')) {
+      throw new NotFoundHttpException();
+    }
+    $langcode = $this->getCurrentLanguageId();
+    $cacheableMetadata = (new CacheableMetadata())
+      ->addCacheableDependency($request->attributes->get(AccessAwareRouterInterface::ACCESS_RESULT))
+      ->addCacheTags([
+        'helfi_menu_link_collection',
+        'config:system.menu.' . $menuName,
+      ]);
+
+    try {
+      $tree = $this->treeBuilder
+        ->build($menuName, $langcode);
+      $response = new ResourceResponse($this->toArray($tree), 200);
+      $response->addCacheableDependency($cacheableMetadata);
+
+      return $response;
+    }
+    catch (\Exception) {
+    }
+    throw new BadRequestHttpException(sprintf('Failed to load menu: %s', $menuName));
+  }
+
+  /**
+   * Converts menu tree objects to array.
+   *
+   * @param \stdClass[] $items
+   *   The items.
+   *
+   * @return array
+   *   The response converted to array.
+   */
+  private function toArray(array $items) : array {
+    // The menu tree builder returns an array of stdClass objects, but
+    // Drupal doesn't know how to normalize them. Creating a generic
+    // normalizer for stdClass might have some unintended consequences, so
+    // we "normalize" them by converting them to json and back to array.
+    $json = \GuzzleHttp\json_encode($items);
```

Is there a reason to use GuzzleHttp json_encode instead of just php native json_encode? And same for decode. The main difference seems to be that the GuzzleHttp version throws instead of returning false. Is that intentional? 
edit: The php native one has a flag for JSON_THROW_ON_ERROR.